### DISCUSSION TITLE
Disable saving relation defined in fieldset on null values.

### DISCUSSION THREAD

**Discussion by shirokovnv** [filamentphp/filament#6732] [Help] [answered]
Hello, guys!

I have the following database structure:

- article (id, title, body)
- image (polymorphic relation with imageable_id, imageable_type, path, title).

in Article Resource I have defined 

```
Forms\Components\Fieldset::make('image')
                                    ->relationship('image')
                                    ->schema([
                                        Forms\Components\FileUpload::make('path')
                                            ->image()
                                            ->nullable(),
                                        Forms\Components\TextInput::make('title')
                                            ->nullable()
                                    ]),
```

The problem is when I press the save button, If the path and title of the image are not specified, I encountering SQL error: 

NOT NULL CONSTRAINT VIOLATION: path cannot be null in image table.

My question: Is it possible to add some kind of condition to not save a relation, when fields are nulls ?

I will appreciate any help! 

Thanks.

**Comment by danharrin** [comment, marked as answer]
Yeah `relationship(condition: fn ($state) => count(array_filter($state)))`

**Comment by shirokovnv** [comment]
Thanks a lot!
